app/controllers/projects_controller/validators.rb

```diff
@@ -32,4 +32,12 @@ def valid_dates
     flash_error(:add_project_ends_before_start.t)
     false
   end
+
+  def check_constraint_violations
+    count = @project.count_violations
+    return unless count.positive? && @is_admin
```

Comment:
@mo-nathan: Note: this warning goes only to admins

Reply:
That seems reasonable and makes me think we should check @is_admin before evaluating @project.count_violations.  So add an initial check of `return unless @is_admin`.